include/GIL/BasicBlock.hpp

```diff
@@ -0,0 +1,73 @@
+#ifndef GLU_GIL_BASICBLOCK_HPP
+#define GLU_GIL_BASICBLOCK_HPP
+
+#include <llvm/ADT/ilist.h>
+#include <llvm/ADT/ilist_node.h>
+#include <string>
+
+namespace glu::gil {
+class BasicBlock;
+class InstBase : public llvm::ilist_node_with_parent<InstBase, BasicBlock> {
+    BasicBlock *parent = nullptr;
+    friend class BasicBlock; // Allow BasicBlock to set itself as
+public:
+    InstBase() {};
+    virtual ~InstBase() = default;
+    InstBase(InstBase const &) = default;
+    BasicBlock *getParent() const
+    {
+        return parent;
+    }
+    BasicBlock *getParent()
+    {
+        return parent;
+    }
+};
+class TerminatorInst : public InstBase { };
+
+/**
+ * @class BasicBlock
+ * @brief Represents a basic block for instructions in the GIL (Glu Intermediate
+ * Language).
+ * See the documentation here for more information:
+ * https://glu-lang.org/gil/#basic-blocks
+ */
+class BasicBlock {
+public:
+    BasicBlock();
+    BasicBlock(std::string label);
```

Comment:
the constructor is small enough, you can inline it (and combine both)
```suggestion
    BasicBlock(std::string label = ""): _label(label) { }
```